crates/runtime/src/dataaccelerator/duckdb.rs

```diff
@@ -89,14 +106,66 @@ impl DataAccelerator for DuckDBAccelerator {
         "duckdb"
     }
 
+    async fn init(
+        &self,
+        dataset: &Dataset,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let path = self.duckdb_file_path(dataset);
+
+        if let (Some(path), Some(acceleration)) = (&path, &dataset.acceleration) {
+            if !acceleration.params.contains_key("duckdb_open") {
+                make_spice_data_directory().map_err(|err| {
+                    Error::AccelerationInitializationFailed { source: err.into() }
+                })?;
+            }
+
+            DuckDbConnectionPool::new_file(path, &AccessMode::ReadWrite)
+                .context(AccelerationInitializationFailedSnafu)?;
+        }
+
+        Ok(())
+    }
+
     /// Creates a new table in the accelerator engine, returning a `TableProvider` that supports reading and writing.
     async fn create_external_table(
         &self,
         cmd: &CreateExternalTable,
-        _dataset: Option<&Dataset>,
+        dataset: Option<&Dataset>,
     ) -> Result<Arc<dyn TableProvider>, Box<dyn std::error::Error + Send + Sync>> {
+        let mut cmd = cmd.clone();
+        if !cmd.options.contains_key("open") {
+            make_spice_data_directory()
+                .map_err(|err| Error::AccelerationCreationFailed { source: err.into() })?;
+        }
+
+        if let Some(Some(attach_databases)) = dataset.map(|this_dataset: &Dataset| {
+            this_dataset.app.as_ref().map(|a| {
+                let datasets = Runtime::get_valid_datasets(a, crate::LogErrors(false));
+                datasets
+                    .iter()
+                    .filter(|d| {
+                        d.acceleration
+                            .as_ref()
+                            .map_or(false, |a| a.engine == Engine::DuckDB)
+                    })
+                    .filter_map(|other_dataset| {
+                        if **other_dataset == *this_dataset {
+                            None
+                        } else {
+                            self.duckdb_file_path(other_dataset)
+                        }
+                    })
+                    .collect::<Vec<String>>()
+            })
+        }) {
+            if !attach_databases.is_empty() {
+                cmd.options
+                    .insert("attach_databases".to_string(), attach_databases.join(";"));
+            }
+        }
```

Comment:
the attach databases only inits during the create? If so, when podwatcher updates/creates datasets, they won't be attached with each other even they can be?

Reply:
I've [raised an issue](https://github.com/spiceai/spiceai/issues/2339) to track some work on this. There's an underlying bug where all datasets refresh on any dataset change, although we should consider how we want new datasets to behave with attachments.